### PR TITLE
1810 - Tony Robertson

### PR DESCRIPTION
**Pull request by tonyr729**

### CODE REVIEW

File: lib/house.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    @rooms << room
+  end
+
+  def rooms_from_category(category)
+    @rooms.select do |room|
+      room.category == category
+    end
+  end
+
+  def area
```
Comment (memcmahon) [inline lib/house.rb]: This is great - one interesting thing about line 26 is that the return value is the same as the current value of `total_area` so you can get away with not including row 27 :)

File: lib/house.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    (price.to_f / area.to_f).round(2)
+  end
+
+  def rooms_sorted_by_area
+    @rooms.sort_by do |room|
+      room.area
+    end.reverse
+  end
+
+  def rooms_by_category
```
Comment (memcmahon) [inline lib/house.rb]: Look into the method `group_by` - it could help clean this method up a bit!

File: lib/room.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+class Room
+  attr_reader :category
+  def initialize(category, height, width)
+    @category = category
+    @height = height
```
Comment (memcmahon) [inline lib/room.rb]: I would not be opposed to just calculating the area as an instance variable if you wanted to tighten this class up just a bit.

File: test/house_test.rb [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+  def test_it_has_an_address
+    assert_equal "123 sugar lane", @house.address
+  end
+
+  def test_it_starts_with_no_rooms
+    assert_equal [], @house.rooms
+  end
+
+  def test_it_can_add_rooms
```
Comment (memcmahon) [inline test/house_test.rb]: We want to add multiple rooms to fully test this add_rooms method - want to make sure it works not just once, but always.

File: lib/house.rb (hunk 2 above)
Comment (tonyr729) [reply lib/house.rb]: Yeah, I knew there was a method that does this. Forgot it and just decided to code it out. 

This is my biggest area of improvement I feel. So used to just having to code out things like this. It comes automatically.

File: test/house_test.rb (hunk 4 above)
Comment (tonyr729) [reply test/house_test.rb]: 👍 Noted

File: lib/room.rb (hunk 3 above)
Comment (tonyr729) [reply lib/room.rb]: Mmm ok. I was going off the rule I've heard **"Avoid instance varibles as much as possible"**

File: lib/house.rb (hunk 1 above)
Comment (tonyr729) [reply lib/house.rb]: Ahh, I see. Still weirded out by the implicit returns.

**Pull request closed without merging**